clients/clients-integration-tests/src/test/java/org/apache/kafka/clients/consumer/PlaintextConsumerPollTest.java

```diff
@@ -0,0 +1,686 @@
+        testNoOffsetForPartitionExceptionOnPollZero(GroupProtocol.CLASSIC);
+    }
+
+    @ClusterTest
+    public void testAsyncConsumerNoOffsetForPartitionExceptionOnPollZero() throws InterruptedException {
+        testNoOffsetForPartitionExceptionOnPollZero(GroupProtocol.CONSUMER);
+    }
+
+    private void testNoOffsetForPartitionExceptionOnPollZero(GroupProtocol groupProtocol) throws InterruptedException {
+        Map<String, Object> config = Map.of(GROUP_PROTOCOL_CONFIG, groupProtocol.name().toLowerCase(Locale.ROOT), ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "none");
+        try (Consumer<byte[], byte[]> consumer = cluster.consumer(config)) {
+            consumer.assign(List.of(tp));
+
+            // continuous poll should eventually fail because there is no offset reset strategy set 
+            // (fail only when resetting positions after coordinator is known)
+            TestUtils.waitForCondition(() -> {
+                try {
+                    assertThrows(NoOffsetForPartitionException.class, () -> consumer.poll(Duration.ZERO));
+                    return true;
+                } catch (AssertionError e) {
+                    // ignore
+                }
+                return false;
+            }, "Continuous poll not fail");
+        }
+    }
+
+    private void awaitRebalance(
+        Consumer<byte[], byte[]> consumer, 
+        TestConsumerReassignmentListener rebalanceListener
+    ) throws InterruptedException {
+        var numReassignments = rebalanceListener.callsToAssigned;
+        TestUtils.waitForCondition(() -> {
+                consumer.poll(Duration.ofMillis(100));
+                return rebalanceListener.callsToAssigned > numReassignments;
+            }, "Timed out before expected rebalance completed"
+        );
+    }
+
+    private void ensureNoRebalance(
+        Consumer<byte[], byte[]> consumer, 
+        TestConsumerReassignmentListener rebalanceListener
+    ) throws InterruptedException {
+        // The best way to verify that the current membership is still active is to commit offsets.
+        // This would fail if the group had rebalanced.
+        var initialRevokeCalls = rebalanceListener.callsToRevoked;
+        sendAndAwaitAsyncCommit(consumer);
+        assertEquals(initialRevokeCalls, rebalanceListener.callsToRevoked);
+    }
+
+    private void sendAndAwaitAsyncCommit(
+        Consumer<byte[], byte[]> consumer
+    ) throws InterruptedException {
+        java.util.function.Consumer<OffsetCommitCallback> sendAsyncCommit = consumer::commitAsync;
+
+        var commitCallback = new RetryCommitCallback(sendAsyncCommit);
+        sendAsyncCommit.accept(commitCallback);
+        TestUtils.waitForCondition(() -> {
+                consumer.poll(Duration.ofMillis(100));
+                return commitCallback.isComplete;
+            }, 10000, "Failed to observe commit callback before timeout"
+        );
+        assertTrue(commitCallback.error.isEmpty());
```

Comment:
Could you please use `assertEquals(Optional.empty(), commitCallback.error);` instead? That can show the error message if it fails